src/Interpreters/Context.cpp

```diff
@@ -1549,6 +1549,29 @@ void Context::initializeSystemLogs()
     shared->system_logs.emplace(*global_context, getConfigRef());
 }
 
+void Context::createSystemLogs()
+{
+    auto system_logs = std::make_optional<SystemLogs>(*global_context, getConfigRef());
+
+    int type = SystemLogType::UNDEFINED;
+    if (system_logs->query_log)
+        type |= SystemLogType::QUERY_LOG;
+    if (system_logs->query_thread_log)
+        type |= SystemLogType::QUERY_THREAD_LOG;
+    if (system_logs->part_log)
+        type |= SystemLogType::PART_LOG;
+    if (system_logs->trace_log)
+        type |= SystemLogType::TRACE_LOG;
+    if (system_logs->text_log)
+        type |= SystemLogType::TEXT_LOG;
+    if (system_logs->metric_log)
+        type |= SystemLogType::METRIC_LOG;
+    system_logs->initializeSystemLogs(type);
+
+    auto lock = getLock();
+    shared->system_logs.swap(system_logs);
```

Comment:
`clang` [reports](https://clickhouse-builds.s3.yandex.net/9642/2ef09b86a489e60abc0dbac4a78d1c75accba5b3/build_log_649345256_1586449059.txt):

```
2020-04-09 15:42:58 In file included from ../src/Interpreters/PartLog.h:3:
2020-04-09 15:42:58 ../src/Interpreters/SystemLog.h:86:5: error: definition of implicit copy constructor for 'SystemLogs' is deprecated because it has a user-declared destructor [-Werror,-Wdeprecated]
2020-04-09 15:42:58     ~SystemLogs();
2020-04-09 15:42:58     ^
2020-04-09 15:42:58 ../contrib/libcxx/include/type_traits:3695:9: note: in implicit copy constructor for 'DB::SystemLogs' first required here
2020-04-09 15:42:58     _Tp __t(_VSTD::move(__x));
2020-04-09 15:42:58         ^
2020-04-09 15:42:58 ../contrib/libcxx/include/optional:857:17: note: in instantiation of function template specialization 'std::__1::swap<DB::SystemLogs>' requested here
2020-04-09 15:42:58                 swap(this->__get(), __opt.__get());
2020-04-09 15:42:58                 ^
2020-04-09 15:42:58 ../src/Interpreters/Context.cpp:1570:25: note: in instantiation of member function 'std::__1::optional<DB::SystemLogs>::swap' requested here
2020-04-09 15:42:58     shared->system_logs.swap(system_logs);
```

I don't see any option except for adding default copy/assigment, i.e.:

```patch
diff --git a/src/Interpreters/SystemLog.h b/src/Interpreters/SystemLog.h
index 87da342ae1..5b3a3fcba2 100644
--- a/src/Interpreters/SystemLog.h
+++ b/src/Interpreters/SystemLog.h
@@ -72,6 +72,9 @@ class MetricLog;
 struct SystemLogs
 {
     SystemLogs(Context & global_context, const Poco::Util::AbstractConfiguration & config);
+    SystemLogs(const SystemLogs &) = default;
+    SystemLogs& operator=(const SystemLogs &) = default;
+
     ~SystemLogs();
 
     void shutdown();
```